File: rules/gitlab/java/inject/test-ELInjection.java
// License: LGPL-3.0 License (c) find-sec-bugs
import javax.el.*;
import javax.faces.context.FacesContext;

public class ELInjection {

    public String valueExpr(String expression) {
        FacesContext context = FacesContext.getCurrentInstance();
        ExpressionFactory expressionFactory = context.getApplication().getExpressionFactory();
        ELContext elContext = context.getELContext();
        ValueExpression vex = expressionFactory.createValueExpression(elContext, expression, String.class);
        return (String) vex.getValue(elContext);
    }

    public MethodInfo methodExpr(String expression) {
        FacesContext context = FacesContext.getCurrentInstance();
        ExpressionFactory expressionFactory = context.getApplication().getExpressionFactory();
        ELContext elContext = context.getELContext();
        MethodExpression ex = expressionFactory.createMethodExpression(elContext, expression, String.class, null);
        return ex.getMethodInfo(elContext);
    }
    
}

File: rules/gitlab/scala/inject/test-ELInjection.scala
// License: LGPL-3.0 License (c) find-sec-bugs
package inject

import javax.el._
import javax.faces.context.FacesContext


class ELInjection {
  def valueExpr(expression: String) = {
    val context = FacesContext.getCurrentInstance
    val expressionFactory = context.getApplication.getExpressionFactory
    val elContext = context.getELContext
    val vex = expressionFactory.createValueExpression(elContext, expression, classOf[Nothing])
    vex.getValue(elContext).asInstanceOf[Nothing]
  }

  def methodExpr(expression: String) = {
    val context = FacesContext.getCurrentInstance
    val expressionFactory = context.getApplication.getExpressionFactory
    val elContext = context.getELContext
    val ex = expressionFactory.createMethodExpression(elContext, expression, classOf[Nothing], null)
    ex.getMethodInfo(elContext)
  }
}
